File: JAVA/TrigoNoTabuleiroTest.java
/** 
 * @version JAVA
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ mvn clean test -Dtest=your.package.TestClassName
 */ 
package JAVA;

import static org.junit.Assert.assertEquals;
import org.junit.Test;
import java.math.BigInteger;

public class TrigoNoTabuleiroTest {

	@Test
	public void testCornToReceive() throws Exception {
		//Test 1
		BigInteger want = BigInteger.valueOf(0);
		BigInteger got = TrigoNoTabuleiro.cornToReceive(0);
		assertEquals(want, got);
		//Test 2
		want = BigInteger.valueOf(0);
		got = TrigoNoTabuleiro.cornToReceive(8);
		assertEquals(want, got);
		//Test 3
		want = BigInteger.valueOf(5);
		got = TrigoNoTabuleiro.cornToReceive(16);
		assertEquals(want, got);
		//Test 4
		want = BigInteger.valueOf(1398);
		got = TrigoNoTabuleiro.cornToReceive(24);
		assertEquals(want, got);
		//Test 5
		want = BigInteger.valueOf(357913);
		got = TrigoNoTabuleiro.cornToReceive(32);
		assertEquals(want, got);
		//Test 6
		want = BigInteger.valueOf(91625968);
		got = TrigoNoTabuleiro.cornToReceive(40);
		assertEquals(want, got);
		//Test 7
		want = new BigInteger("23456248059");
		got = TrigoNoTabuleiro.cornToReceive(48);
		assertEquals(want, got);
		//Test 8
		want = new BigInteger("6004799503160");
		got = TrigoNoTabuleiro.cornToReceive(56);
		assertEquals(want, got);
		//Test 9
		want = new BigInteger("1537228672809129");
		got = TrigoNoTabuleiro.cornToReceive(64);
		assertEquals(want, got);
	}
}

File: JS/TrigoNoTabuleiro.test.js
/**
 * @version JAVASCRIPT
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ node NomeDaClasse.test
 */
const TEST = require('tape')
const OBJ = require('./TrigoNoTabuleiro.js')

TEST('Starting trigo_no_tabuleiro test...', (t) => {
    //Test 1
    t.assert(OBJ.cornToReceive(0) === 0, "Expect: 0 = 0 kg")
    //Test 2
    t.assert(OBJ.cornToReceive(8) === 0, "Expect: 8 = 0 kg")
    //Test 3
    t.assert(OBJ.cornToReceive(16) === 5, "Expect: 16 = 5 kg")
    //Test 4
    t.assert(OBJ.cornToReceive(24) === 1398, "Expect: 24 = 1398 kg")
    //Test 5
    t.assert(OBJ.cornToReceive(32) === 357913, "Expect: 32 = 357913 kg")
    //Test 6
    t.assert(OBJ.cornToReceive(40) === 91625968, "Expect: 40 = 91625968 kg")
    //Test 7
    t.assert(OBJ.cornToReceive(48) === 23456248059, "Expect: 48 = 23456248059 kg")
    //Test 8
    t.assert(OBJ.cornToReceive(56) === 6004799503160, "Expect: 56 = 6004799503160 kg")
    //Test 9
    t.assert(OBJ.cornToReceive(64) === 1537228672809129, "Expect: 64 = 1537228672809129 kg")

    t.end()
})
 
